src/sidebar/util/url.js

```diff
@@ -1,3 +1,8 @@
+/**
+ * @typedef {string|number|boolean} Param
+ * @typedef {Record<string, Param|Param[]>} Params
+ */
```

Comment:
These types are conveniently the same as those in `services/api.js`. I didn't import those here to preserve the dependency structure (`services` may depend on `util` but not vice-versa). Instead I just defined conveniently-equivalent types.